### ISSUE TITLE
Detect C lang printing functions and suggest the appropriate macro call

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust#92898]
We already translate from C-style formatting strings to Rust formatting strings when passed to the `format!` family of macros. We should also detect invalid `fn` calls from other languages that were meant to be macro calls:

```rust
fn main() {
    let x = 4;
    printf("%d", x);
}
```

currently emits the dry

```
error[E0425]: cannot find function `printf` in this scope
 --> <source>:3:5
  |
3 |     printf("%d", x);
  |     ^^^^^^ not found in this scope
```

while it could be

```
error[E0425]: cannot find function `printf` in this scope
 --> <source>:3:5
  |
3 |     printf("%d", x);
  |     ^^^^^^ not found in this scope
help: you might have meant to use the `print!` macro
 --> <source>:3:5
  |
3 |     print!("{}", x);
  |     ~~~~~~  ~~
```

**Comment by vincenzopalazzo** [comment]
Mentoring @amustaque97 on this issue

@rustbot label +E-mentor
